Source/WebCore/css/StyleProperties.cpp

```diff
@@ -575,7 +576,87 @@ static std::optional<CSSValueID> fontStretchKeyword(double value)
 
 String StyleProperties::fontValue() const
 {
-    // FIXME: This should check more font subproperties aand return empty string when they do not have their initial values. https://bugs.webkit.org/show_bug.cgi?id=247498
+    auto propertyReferenceKeywordValue = [&](const PropertyReference& property) {
+        auto* value = dynamicDowncast<CSSPrimitiveValue>(property.value());
+        return value ? value->valueID() : CSSValueInvalid;
+    };
+
+    auto propertyKeywordValue = [&](CSSPropertyID property) -> std::optional<CSSValueID> {
+        if (int foundPropertyIndex = findPropertyIndex(property); foundPropertyIndex != -1)
+            return propertyReferenceKeywordValue(propertyAt(foundPropertyIndex));
+        return std::nullopt;
+    };
+
+    auto propertyIsDefault = [&](CSSPropertyID property, CSSValueID defaultValue) {
+        auto keyword = propertyKeywordValue(property);
+        return !keyword || *keyword == CSSValueInitial || *keyword == defaultValue;
+    };
+
+    auto propertyReferenceIsCSSWideKeyword = [&](const PropertyReference& property) {
+        return isCSSWideKeyword(propertyReferenceKeywordValue(property));
+    };
+
+    auto propertiesResetByShorthandAreExpressible = [&] {
```

Comment:
Could the font-stretch check go in here as well?

Reply:
The `font-stretch` check happens as a side effect of mapping to a keyword. If I separated the check from the keyword mapping it would be less efficient, so I left it as is for now.